src/edu/washington/escience/myria/parallel/Server.java

```diff
@@ -1033,6 +1034,45 @@ public void importDataset(final RelationKey relationKey, final Schema schema, fi
   }
 
   /**
+   * @param relationKey the relationalKey of the dataset to import
+   * @return the status
+   * @throws DbException if there is an error
+   * @throws InterruptedException interrupted
+   */
+  public DatasetStatus deleteDataset(final RelationKey relationKey) throws DbException, InterruptedException {
+
+    /* delete from postgres at each worker by calling the DbDelete operator */
+    try {
+      Map<Integer, SubQueryPlan> workerPlans = new HashMap<>();
+      for (Integer workerId : getWorkersForRelation(relationKey, null)) {
+        workerPlans.put(workerId, new SubQueryPlan(new DbDelete(EmptyRelation.of(catalog.getSchema(relationKey)),
+            relationKey, null)));
+      }
+      ListenableFuture<Query> qf =
+          queryManager.submitQuery("delete " + relationKey.toString(), "delete " + relationKey.toString(),
```

Comment:
I've never used the two-phase commit either, but I think this comment ties along with what you mentioned at the comment under ```MasterCatalog.java``` (the interaction between the master catalog and what relations the workers currently have -- at creation and deletion).